sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -661,6 +663,68 @@ public CosmosAsyncConflict getConflict(String id) {
                    }).map(offerResourceResponse -> offerResourceResponse.getResource().getThroughput());
     }
 
+    /**
+     * Replace the throughput .
+     *
+     * @param throughputProperties the throughput properties
+     * @return the mono containing throughput response
+     */
+    public Mono<ThroughputResponse> replaceThroughput(ThroughputProperties throughputProperties) {
+        return this.read()
+                   .flatMap(response -> this.database.getDocClientWrapper()
+                                            .queryOffers(database.getOfferQuerySpecFromResourceId(response
+                                                                                                      .getProperties()
+                                                                                                      .getResourceId())
+                                                , new FeedOptions())
+                                            .single()
+                                            .flatMap(offerFeedResponse -> {
+                                                if (offerFeedResponse.getResults().isEmpty()) {
+                                                    return Mono.error(BridgeInternal
+                                                                          .createCosmosClientException(
+                                                                              HttpConstants.StatusCodes.BADREQUEST,
+                                                                              "No offers found for the " +
```

Comment:
In practice its only possible if name is invalid right? 
How about reflecting the same in the exception message.

Reply:
Added resource id in the error message which should help.